hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -369,15 +369,19 @@ def __init__(self,
     async def cancel(self):
         await self._client._patch(f'/api/v1alpha/batches/{self.id}/cancel')
 
-    async def jobs(self, q=None):
+    async def jobs(self, q=None, version=None):
+        if version is None:
+            version = 1
+        if not (isinstance(version, int) and version >= 1):
```

Comment:
This feels a little out of character for the codebase. It seems to me that this sort of user-input validation belongs as close as possible to where the user would provide it instead of deep down in the implementation. I think inside these internal modules we can just add a type signature to this function and rely on that. Then if we expose this option to the user we should do this ValueError check there.

Reply:
I thought we considered `aioclient.py` an internal API, and the user API being the sync client

Reply:
I didn't realize that, I thought we make breaking changes to the aioclient interface.

Reply:
And it is undocumented.

Reply:
I think we should just add types to the signature of this function, so that way any internal uses should fail linting if they don't use integers, and then delete this if statement that's doing validation. The front end should validate the version anyway.